### ISSUE TITLE
CRAB jobwrapper should be python3-compatible

### ISSUE THREAD

**Issue by mapellidario** [dmwm/CRABServer#7210]
Currently, we only run the CRAB jobwrapper in a py2 environment [1].

As an excuse to play around with our jobwrapper to better understand https://github.com/dmwm/CRABServer/issues/7084, I ported to `CMSRunAnalysis.sh` the diffs between WMCore's `submit.sh` and `submit_py3.sh` and tested it after a `crab preparelocal`.

The current jobwrapper CMSRunAnalysis.py is not py3 compatible. for the time being, I found the following problem

- [ ] using getoutput from `commands` [2]. This should be easy to fix, https://python-future.org/compatible_idioms.html#commands-subprocess-modules

#### further info

[1]

https://github.com/dmwm/CRABServer/blob/01c1bdd55853164a55a669cfa581212d17cf5682/scripts/CMSRunAnalysis.sh#L138

[2]

```plaintext
[...]
Now running the CMSRunAnalysis.py job in /afs/cern.ch/user/d/dmapelli/crab/temp/20220411-preparelocal-2...
++ pwd
+ python3 CMSRunAnalysis.py -r /afs/cern.ch/user/d/dmapelli/crab/temp/20220411-preparelocal-2 -a sandbox.tar.gz --sourceURL=https://cmsweb-testbed.cern.ch/S3/crabcache_preprod --jobNumber=1 --cmsswVersion=CMSSW_12_2_1 --scramArch=slc7_amd64_gcc10 --inputFile=job_input_file_list_1.txt --runAndLumis=job_lumis_1.json --lheInputFiles=False --firstEvent=None --firstLumi=None --lastEvent=None --firstRun=None --seeding=AutomaticSeeding --scriptExe=None --eventsPerLumi=None --maxRuntime=-1 '--scriptArgs=[]' -o '{}'
Traceback (most recent call last):
  File "CMSRunAnalysis.py", line 19, in <module>
    import commands
ModuleNotFoundError: No module named 'commands'
+ jobrc=1
+ set +x
== The job had an exit code of 1
[...]
```

**Comment by mapellidario** [comment]
Before I forget again, let me write down something that I already discovered and already forgot.

At the moment, we run by default our jobwrapper with `slc6_amd64_gcc493`, see

https://github.com/dmwm/CRABServer/blob/01c1bdd55853164a55a669cfa581212d17cf5682/scripts/CMSRunAnalysis.sh#L61

If we want to run the jobwrapper with python3, we would need to source a different SCRAM_ARCH, such as WMCore does, see

https://github.com/dmwm/WMCore/blob/741726019a5b369a47c26a37462e2dbf1f278fd6/etc/submit_py3.sh#L144

added with this PR: https://github.com/dmwm/WMCore/pull/11077

Long story short: running the jobwrapper with python3 will be a by-product of better integrating our job wrapper with WMCore one.  Running the jobwrapper in python3 will be part of https://github.com/dmwm/CRABServer/issues/7084, but I would suggest leaving this issue open not to forget to check the python version.

**Comment by belforte** [comment]
can you simply add to https://github.com/dmwm/CRABServer/issues/7084 that new script will run in python3 ?

**Comment by mapellidario** [comment]
I agree. I added it as an action item in this comment https://github.com/dmwm/CRABServer/issues/7084#issuecomment-1192371547, I will close this issue since it is going to be a by-product of #7084 anyways. Thanks Stefano!
